ray-operator/controllers/ray/raycluster_controller.go

```diff
@@ -438,8 +437,7 @@ func (r *RayClusterReconciler) reconcileIngress(ctx context.Context, instance *r
 func (r *RayClusterReconciler) reconcileRouteOpenShift(ctx context.Context, instance *rayv1.RayCluster) error {
 	logger := ctrl.LoggerFrom(ctx)
 	headRoutes := routev1.RouteList{}
-	filterLabels := client.MatchingLabels{utils.RayClusterLabelKey: instance.Name}
-	if err := r.List(ctx, &headRoutes, client.InNamespace(instance.Namespace), filterLabels); err != nil {
+	if err := r.List(ctx, &headRoutes, common.RayClusterAllPodsAssociationOptions(instance).ToListOptions()...); err != nil {
```

Comment:
This is pretty confusing. We use `RayClusterAllPodsAssociationOptions` to list routes.